daemon/api.go

```diff
@@ -1911,10 +1911,15 @@ func changeInterfaces(c *Command, r *http.Request, user *auth.UserState) Respons
 		connRef, err = repo.ResolveConnect(a.Plugs[0].Snap, a.Plugs[0].Name, a.Slots[0].Snap, a.Slots[0].Name)
 		if err == nil {
 			var ts *state.TaskSet
+			affected = snapNamesFromConns([]*interfaces.ConnRef{connRef})
 			summary = fmt.Sprintf("Connect %s:%s to %s:%s", connRef.PlugRef.Snap, connRef.PlugRef.Name, connRef.SlotRef.Snap, connRef.SlotRef.Name)
+			if repo.HasConnection(connRef) {
```

Comment:
I still think Connect itself should be responsible to do this check, otherwise somebody will use it in such a situation that re-runs hooks that might do (currently) the wrong thing

Reply:
The problem is Connect doesn't know about repo (nor have access to ifacemgr) and I'm not sure we want to change it?

Reply:
it can look at conns as everything there does, no?

Reply:
It could indeed.. I somehow subconsciously reserved this for handlers. You're right, thanks.

Reply:
it's not different (I think) than various snapstate task set factories looking at "snaps"